### PR TITLE
settings: add and initialize per-key lighting to a special no-change value

### LINKED ISSUE #2459: LED feature lights/changes the color of the devices
**Information**

- Solaar version (`solaar --version` or `git describe --tags` if cloned from this repository): solaar 1.1.12
- Distribution: arch linux
- Kernel version (ex. `uname -srmo`): `6.8.7-arch1-2`
- Output of `solaar show`: 

<details>

```
Lightspeed Receiver
  Device path  : /dev/hidraw5
  USB id       : 046d:C547
  Serial       : 15031480
  C Pending    : ff
    Firmware   : 04.02.B0009
    Bootloader : 02.09
    Other      : 70.7E
  Has 2 paired device(s) out of a maximum of 2.
  Notifications: wireless, software present (0x000900)
  Device activity counters: 1=60

  1: G915 TKL LIGHTSPEED Wireless RGB Mechanical Gaming Keyboard
     Device path  : None
     WPID         : 408E
     Codename     : G915 TKL
     Kind         : keyboard
     Protocol     : HID++ 4.2
     Report Rate : 1ms
     Serial number: C7FAD1B0
     Model ID:      B35F408EC343
     Unit ID:       C7FAD1B0
                 1: BL1 12.01.B0021
                 3: 
                 0: MPK 14.02.B0022
                 3: 
                 3: 
     The power switch is located on the top right corner.
     Supports 38 HID++ 2.0 features:
         0: ROOT                   {0000} V0     
         1: FEATURE SET            {0001} V0     
         2: DEVICE FW VERSION      {0003} V4     
            Firmware: Bootloader BL1 12.01.B0021 00005E599E54
            Firmware: Other   
            Firmware: Firmware MPK 14.02.B0022 408E210DDD69
            Firmware: Other   
            Firmware: Other   
            Unit ID: C7FAD1B0  Model ID: B35F408EC343  Transport IDs: {'btleid': 'B35F', 'wpid': '408E', 'usbid': 'C343'}
         3: DEVICE NAME            {0005} V0     
            Name: G915 TKL LIGHTSPEED Wireless RGB Mechanical Gaming Keyboard
            Kind: keyboard
         4: WIRELESS DEVICE STATUS {1D4B} V0     
         5: CONFIG CHANGE          {0020} V0     
            Configuration: 11000000000000000000000000000000
         6: DEVICE FRIENDLY NAME   {0007} V0     
            Friendly Name: G915 TKL
         7: BATTERY VOLTAGE        {1001} V3     
            Battery: 90% 4124mV , discharging.
         8: CHANGE HOST            {1814} V1     
            Change Host        : 1:mapuo-homepad
         9: HOSTS INFO             {1815} V1     
            Host 0 (paired): mapuo-homepad
            Host 1 (unpaired): 
        10: RGB EFFECTS            {8071} V0     
            LED Control (saved): Device
            LED Control        : Device
            LEDs Logo (saved): !LEDEffectSetting {ID: 0, color: 16711680, intensity: 0, period: 100, ramp: 0, speed: 0}
            LEDs Logo        : HID++ error {'number': 1, 'request': 2796, 'error': 7, 'params': b'\x00'}
            LEDs Primary        : HID++ error {'number': 1, 'request': 2792, 'error': 7, 'params': b'\x01'}
        11: PER KEY LIGHTING V2    {8081} V2     
            Per-key Lighting (saved): {A:white, B:white, C:white, D:white, E:white, F:white, G:white, H:white, I:white, J:white, K:white, L:white, M:white, N:white, O:white, P:white, Q:white, R:white, S:white, T:white, U:white, V:white, W:white, X:white, Y:white, Z:white, 1:white, 2:white, 3:white, 4:white, 5:white, 6:white, 7:white, 8:white, 9:white, 0:white, ENTER:white, ESC:white, BACKSPACE:white, TAB:white, SPACE:white, -:white, =:white, [:white, \:white, KEY 46:white, ~:white, ;:white, ':white, `:white, ,:white, .:white, /:white, CAPS LOCK:white, F1:white, F2:white, F3:white, F4:white, F5:white, F6:white, F7:white, F8:white, F9:white, F10:white, F11:white, F12:white, PRINT:white, SCROLL LOCK:white, PASTE:white, INSERT:white, HOME:white, PAGE UP:white, DELETE:white, END:white, PAGE DOWN:white, RIGHT:white, LEFT:white, DOWN:white, UP:white, KEY 97:white, COMPOSE:white, POWER:white, KEY 100:white, KEY 101:white, KEY 102:white, KEY 103:white, LEFT CTRL:white, LEFT SHIFT:white, LEFT ALT:white, LEFT WINDOWS:white, RIGHT CTRL:white, RIGHT SHIFT:white, RIGHT ALTGR:white, RIGHT WINDOWS:white, BRIGHTNESS:white, PAUSE:white, MUTE:white, NEXT:white, PREVIOUS:white, LOGO:white}
            Per-key Lighting        : {A:white, B:white, C:white, D:white, E:white, F:white, G:white, H:white, I:white, J:white, K:white, L:white, M:white, N:white, O:white, P:white, Q:white, R:white, S:white, T:white, U:white, V:white, W:white, X:white, Y:white, Z:white, 1:white, 2:white, 3:white, 4:white, 5:white, 6:white, 7:white, 8:white, 9:white, 0:white, ENTER:white, ESC:white, BACKSPACE:white, TAB:white, SPACE:white, -:white, =:white, [:white, \:white, KEY 46:white, ~:white, ;:white, ':white, `:white, ,:white, .:white, /:white, CAPS LOCK:white, F1:white, F2:white, F3:white, F4:white, F5:white, F6:white, F7:white, F8:white, F9:white, F10:white, F11:white, F12:white, PRINT:white, SCROLL LOCK:white, PASTE:white, INSERT:white, HOME:white, PAGE UP:white, DELETE:white, END:white, PAGE DOWN:white, RIGHT:white, LEFT:white, DOWN:white, UP:white, KEY 97:white, COMPOSE:white, POWER:white, KEY 100:white, KEY 101:white, KEY 102:white, KEY 103:white, LEFT CTRL:white, LEFT SHIFT:white, LEFT ALT:white, LEFT WINDOWS:white, RIGHT CTRL:white, RIGHT SHIFT:white, RIGHT ALTGR:white, RIGHT WINDOWS:white, BRIGHTNESS:white, PAUSE:white, MUTE:white, NEXT:white, PREVIOUS:white, LOGO:white}
        12: REPROG CONTROLS V4     {1B04} V4     
            Key/Button Diversion (saved): {Host Switch Channel 1:Diverted, Host Switch Channel 2:Diverted}
            Key/Button Diversion        : {Host Switch Channel 1:Diverted, Host Switch Channel 2:Diverted}
        13: REPORT HID USAGE       {1BC0} V1     
        14: ENCRYPTION             {4100} V0     
        15: KEYBOARD DISABLE BY USAGE {4522} V0     
        16: KEYBOARD LAYOUT 2      {4540} V0     
        17: GKEY                   {8010} V0     
            Divert G and M Keys (saved): False
            Divert G and M Keys        : False
        18: MKEYS                  {8020} V0     
            M-Key LEDs (saved): {M1:False, M2:False, M3:False}
            M-Key LEDs        : {M1:False, M2:False, M3:False}
        19: MR                     {8030} V0     
            MR-Key LED (saved): False
            MR-Key LED        : False
        20: BRIGHTNESS CONTROL     {8040} V0     
            Brightness Control (saved): 50
            Brightness Control        : 50
        21: ONBOARD PROFILES       {8100} V0     
            Device Mode: On-Board
            Onboard Profiles (saved): Profile 2
            Onboard Profiles        : Profile 2
        22: REPORT RATE            {8060} V0     
            Report Rate: 1ms
            Report Rate (saved): 1ms
            Report Rate        : 1ms
        23: DFUCONTROL SIGNED      {00C2} V0     
        24: DFU                    {00D0} V3     
        25: DEVICE RESET           {1802} V0    internal, hidden 
        26: unknown:1803           {1803} V0    internal, hidden 
        27: CONFIG DEVICE PROPS    {1806} V8    internal, hidden 
        28: unknown:1813           {1813} V0    internal, hidden 
        29: OOBSTATE               {1805} V0    internal, hidden 
        30: unknown:1830           {1830} V0    internal, hidden 
        31: unknown:1890           {1890} V9    internal, hidden 
        32: unknown:1891           {1891} V9    internal, hidden 
        33: unknown:18A1           {18A1} V0    internal, hidden 
        34: unknown:1E00           {1E00} V0    hidden 
        35: unknown:1EB0           {1EB0} V0    internal, hidden 
        36: unknown:1861           {1861} V0    internal, hidden 
        37: unknown:18B0           {18B0} V0    internal, hidden 
     Has 2 reprogrammable keys:
         0: Host Switch Channel 1     , default: HostSwitch Channel 1        => HostSwitch Channel 1      
             divertable, persistently divertable, pos:1, group:0, group mask:empty
             reporting: diverted
         1: Host Switch Channel 2     , default: HostSwitch Channel 2        => HostSwitch Channel 2      
             divertable, persistently divertable, pos:2, group:0, group mask:empty
             reporting: diverted
     Battery: 90% 4124mV , discharging.

  2: G502 X PLUS
     Device path  : None
     WPID         : 4099
     Codename     : G502 X PLUS
     Kind         : mouse
     Protocol     : HID++ 4.2
     Report Rate : 1ms
     Serial number: 55E91408
     Model ID:      4099C0950000
     Unit ID:       55E91408
                 1: BL1 42.00.B0016
                 0: MPM 27.00.B0016
                 3: 
                 3: 
                 3: 
                 3: 
                 3: 
                 3: 
                 3: 
                 3: 
                 3: 
                 3: 
     Supports 32 HID++ 2.0 features:
         0: ROOT                   {0000} V0     
         1: FEATURE SET            {0001} V0     
         2: DEVICE FW VERSION      {0003} V4     
            Firmware: Bootloader BL1 42.00.B0016 AB0BFBB13A33
            Firmware: Firmware MPM 27.00.B0016 4099FBB13A33
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Firmware: Other   
            Unit ID: 55E91408  Model ID: 4099C0950000  Transport IDs: {'wpid': '4099', 'usbid': 'C095'}
         3: DEVICE NAME            {0005} V0     
            Name: G502 X PLUS
            Kind: mouse
         4: WIRELESS DEVICE STATUS {1D4B} V0     
         5: CONFIG CHANGE          {0020} V0     
            Configuration: 11000000000000000000000000000000
         6: UNIFIED BATTERY        {1004} V3     
            Battery: 95%, discharging.
         7: ADJUSTABLE DPI         {2201} V2     
            Sensitivity (DPI) (saved): 2300
            Sensitivity (DPI)        : 2300
         8: HIRES WHEEL            {2121} V0     
            Multiplier: 8
            Has invert: Normal wheel motion
            Has ratchet switch: Normal wheel mode
            Low resolution mode
            HID notification
            Scroll Wheel Direction (saved): False
            Scroll Wheel Direction        : False
            Scroll Wheel Resolution (saved): False
            Scroll Wheel Resolution        : False
            Scroll Wheel Diversion (saved): False
            Scroll Wheel Diversion        : False
         9: RGB EFFECTS            {8071} V2     
            LED Control (saved): Device
            LED Control        : Device
            LEDs Primary        : HID++ error {'number': 2, 'request': 2543, 'error': 7, 'params': b'\x00'}
        10: PER KEY LIGHTING V2    {8081} V2     
            Per-key Lighting (saved): {A:white, B:white, C:white, D:white, E:white, F:white, G:white, H:white}
            Per-key Lighting        : {A:white, B:white, C:white, D:white, E:white, F:white, G:white, H:white}
        11: ONBOARD PROFILES       {8100} V0     
            Device Mode: On-Board
            Onboard Profiles (saved): Profile 1
            Onboard Profiles        : Profile 1
        12: MOUSE BUTTON SPY       {8110} V0     
        13: REPORT RATE            {8060} V0     
            Report Rate: 1ms
            Report Rate (saved): 1ms
            Report Rate        : 1ms
        14: FORCE PAIRING          {1500} V0     
        15: DFU                    {00D0} V3     
        16: DEVICE RESET           {1802} V0    internal, hidden, unknown:000010 
        17: unknown:1803           {1803} V0    internal, hidden, unknown:000010 
        18: CONFIG DEVICE PROPS    {1806} V8    internal, hidden, unknown:000010 
        19: unknown:1811           {1811} V0    internal, hidden, unknown:000010 
        20: OOBSTATE               {1805} V0    internal, hidden 
        21: unknown:1830           {1830} V0    internal, hidden, unknown:000010 
        22: unknown:1875           {1875} V0    internal, hidden, unknown:000010 
        23: unknown:1861           {1861} V1    internal, hidden, unknown:000010 
        24: unknown:1890           {1890} V7    internal, hidden, unknown:000008 
        25: unknown:18A1           {18A1} V0    internal, hidden, unknown:000010 
        26: unknown:1801           {1801} V0    internal, hidden, unknown:000010 
        27: unknown:1E00           {1E00} V0    hidden 
        28: unknown:1E22           {1E22} V0    internal, hidden, unknown:000010 
        29: unknown:1EB0           {1EB0} V0    internal, hidden, unknown:000010 
        30: unknown:18B1           {18B1} V0    internal, hidden, unknown:000010 
        31: unknown:18C0           {18C0} V0    internal, hidden, unknown:000010 
     Battery: 95%, discharging.

```
</details>

- Contents of `~/.config/solaar/config.yaml` (or `~/.config/solaar/config.json` if `~/.config/solaar/config.yaml` not present):

<details>

```
- 1.1.12
- _NAME: MX Vertical Wireless Mouse
  _modelId: B020407BC08A
  _sensitive: {dpi: false}
  _serial: 3A83FBA3
  _unitId: 1CF35693
  _wpid: 407B
  change-host: null
  divert-keys: {82: 0, 83: 0, 86: 0, 253: 0}
  dpi: 2100
  lowres-scroll-mode: false
  lowres-smooth-scroll: false
  persistent-remappable-keys: null
  reprogrammable-keys: {80: 80, 81: 81, 82: 82, 83: 83, 86: 86, 253: 253}
- _NAME: MX Master 3 Wireless Mouse
  _modelId: B02340820000
  _sensitive: {dpi: false, smart-shift: false, thumb-scroll-invert: false}
  _serial: FB0615BE
  _unitId: FB0615BE
  _wpid: '4082'
  change-host: null
  divert-keys: {82: 0, 83: 0, 86: 0, 195: 0, 196: 0}
  dpi: 2300
  hires-scroll-mode: false
  hires-smooth-invert: false
  hires-smooth-resolution: true
  reprogrammable-keys: {80: 80, 81: 81, 82: 82, 83: 83, 86: 86, 195: 195, 196: 196}
  scroll-ratchet: 2
  smart-shift: 20
  thumb-scroll-invert: true
  thumb-scroll-mode: false
- _NAME: Wireless Mouse M185 new
  _modelId: '000000000000'
  _sensitive: {hires-scroll-mode: ignore, hires-smooth-invert: ignore, hires-smooth-resolution: ignore}
  _serial: '00000000'
  _unitId: '00000000'
  _wpid: '4054'
  divert-keys: {82: 0}
  lowres-scroll-mode: false
  pointer_speed: 256
  reprogrammable-keys: {80: 80, 81: 81, 82: 82}
- _NAME: MX Keys Keyboard
  _sensitive: {disable-keyboard-keys: false, divert-keys: false, multiplatform: false}
  _serial: 18DC7451
  _wpid: 408A
  backlight: true
  change-host: null
  disable-keyboard-keys: {1: false, 2: false, 4: false, 8: false, 16: false}
  divert-keys: {10: 0, 110: 0, 111: 0, 191: 0, 199: 0, 200: 0, 209: 0, 210: 0, 211: 0, 224: 0, 225: 0, 226: 0, 227: 0, 228: 0, 229: 0, 230: 0, 231: 0, 232: 0,
    233: 0, 234: 0, 235: 0, 236: 0}
  fn-swap: false
  multiplatform: 0
- _NAME: G915 TKL LIGHTSPEED Wireless RGB Mechanical Gaming Keyboard
  _absent: [hi-res-scroll, lowres-scroll-mode, hires-smooth-invert, hires-smooth-resolution, hires-scroll-mode, scroll-ratchet, smart-shift, thumb-scroll-invert,
    thumb-scroll-mode, report_rate_extended, pointer_speed, dpi, dpi_extended, speed-change, backlight, backlight_level, backlight_duration_hands_out, backlight_duration_hands_in,
    backlight_duration_powered, backlight-timed, led_control, led_zone_, fn-swap, reprogrammable-keys, persistent-remappable-keys, disable-keyboard-keys,
    crown-smooth, divert-crown, multiplatform, gesture2-gestures, gesture2-divert, gesture2-params, sidetone, equalizer, adc_power_management]
  _battery: 4097
  _modelId: B35F408EC343
  _sensitive: {divert-gkeys: true, divert-keys: false, rgb_control: false, rgb_zone_1: false, rgb_zone_2: false}
  _serial: C7FAD1B0
  _unitId: C7FAD1B0
  _wpid: 408E
  brightness_control: 50
  change-host: null
  divert-gkeys: false
  divert-keys: {209: 1, 210: 1}
  m-key-leds: {1: false, 2: false, 4: false}
  mr-key-led: false
  onboard_profiles: 2
  per-key-lighting: {1: 16777215, 2: 16777215, 3: 16777215, 4: 16777215, 5: 16777215, 6: 16777215, 7: 16777215, 8: 16777215, 9: 16777215, 10: 16777215,
    11: 16777215, 12: 16777215, 13: 16777215, 14: 16777215, 15: 16777215, 16: 16777215, 17: 16777215, 18: 16777215, 19: 16777215, 20: 16777215, 21: 16777215,
    22: 16777215, 23: 16777215, 24: 16777215, 25: 16777215, 26: 16777215, 27: 16777215, 28: 16777215, 29: 16777215, 30: 16777215, 31: 16777215, 32: 16777215,
    33: 16777215, 34: 16777215, 35: 16777215, 36: 16777215, 37: 16777215, 38: 16777215, 39: 16777215, 40: 16777215, 41: 16777215, 42: 16777215, 43: 16777215,
    44: 16777215, 45: 16777215, 46: 16777215, 47: 16777215, 48: 16777215, 49: 16777215, 50: 16777215, 51: 16777215, 52: 16777215, 53: 16777215, 54: 16777215,
    55: 16777215, 56: 16777215, 57: 16777215, 58: 16777215, 59: 16777215, 60: 16777215, 61: 16777215, 62: 16777215, 63: 16777215, 64: 16777215, 65: 16777215,
    66: 16777215, 67: 16777215, 68: 16777215, 69: 16777215, 70: 16777215, 71: 16777215, 72: 16777215, 73: 16777215, 74: 16777215, 75: 16777215, 76: 16777215,
    77: 16777215, 78: 16777215, 79: 16777215, 97: 16777215, 98: 16777215, 99: 16777215, 100: 16777215, 101: 16777215, 102: 16777215, 103: 16777215, 104: 16777215,
    105: 16777215, 106: 16777215, 107: 16777215, 108: 16777215, 109: 16777215, 110: 16777215, 111: 16777215, 153: 16777215, 155: 16777215, 156: 16777215,
    157: 16777215, 158: 16777215, 210: 16777215}
  report_rate: 1
  rgb_control: 0
  rgb_zone_2: !LEDEffectSetting {ID: 0, color: 16711680, intensity: 0, period: 100, ramp: 0, speed: 0}
- _NAME: G305 Lightspeed Wireless Gaming Mouse
  _modelId: '407400000000'
  _sensitive: {dpi: false}
  _serial: D2717600
  _unitId: 42CE6EF1
  _wpid: '4074'
  dpi: 2300
  onboard_profiles: 1
  report_rate: 1
- _NAME: G502 X PLUS
  _absent: [hi-res-scroll, lowres-scroll-mode, scroll-ratchet, smart-shift, thumb-scroll-invert, thumb-scroll-mode, report_rate_extended, pointer_speed,
    dpi_extended, speed-change, backlight, backlight_level, backlight_duration_hands_out, backlight_duration_hands_in, backlight_duration_powered, backlight-timed,
    led_control, led_zone_, brightness_control, fn-swap, reprogrammable-keys, persistent-remappable-keys, divert-keys, disable-keyboard-keys, crown-smooth,
    divert-crown, divert-gkeys, m-key-leds, mr-key-led, multiplatform, change-host, gesture2-gestures, gesture2-divert, gesture2-params, sidetone, equalizer,
    adc_power_management]
  _battery: 4100
  _modelId: 4099C0950000
  _sensitive: {dpi: true, hires-scroll-mode: true, hires-smooth-invert: true, hires-smooth-resolution: true, per-key-lighting: false, report_rate: true,
    rgb_control: true, rgb_zone_1: false}
  _serial: 55E91408
  _unitId: 55E91408
  _wpid: '4099'
  dpi: 2300
  hires-scroll-mode: false
  hires-smooth-invert: false
  hires-smooth-resolution: false
  onboard_profiles: 1
  per-key-lighting: {1: 16777215, 2: 16777215, 3: 16777215, 4: 16777215, 5: 16777215, 6: 16777215, 7: 16777215, 8: 16777215}
  report_rate: 1
  rgb_control: 0
- _NAME: Candy companion chip
  _modelId: 405F00000000
  _sensitive: {hires-scroll-mode: ignore, hires-smooth-invert: ignore, hires-smooth-resolution: ignore}
  _serial: 458E99C9
  _unitId: '32314715'
  _wpid: 405F

```
</details>

- Errors or warrnings from Solaar:

**Describe the bug**

When Solaar is started the lights of my mouse and keyboard is turned on. They should be off as this is the way I've configured them. I don't want Solaar to control them.

**To Reproduce**
1. Go to Windows Logitech G Hub and disable the lights of the mouse / set your keyboard leds and save them on board
2. Go back to Linux and Solaar will disregard all of those settings and light up the LEDs anyway

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by pfps**
Fixes #2459

### CODE REVIEW

File: lib/logitech_receiver/settings_templates.py [hunk 1]
```diff
@@ -1646,15 +1646,15 @@ class PerKeyLighting(_Settings):
...
+    choices_universe = _special_keys.COLORSPLUS
 
     def read(self, cached=True):
         self._pre_read(cached)
         if cached and self._value is not None:
             return self._value
         reply_map = {}
         for key in self._validator.choices:
-            reply_map[int(key)] = 0xFFFFFF  # can't read so fake a value of white
+            reply_map[int(key)] = -1  # this signals no change
```
Comment (MattHag) [inline lib/logitech_receiver/settings_templates.py:1657]: Please use at least a const with a name. This solves many issues at once.

`NO_KEYLIGHTING = -1`

**Comment by pfps** [comment]
@MattHag The right way to do all the feature request calls is to have an abstract interface for each feature that instead of providing magic symbols for the magic numbers provides function calls with named (and maybe typed) arguments.   But then there is the right way to control a device, which should be done using DBus messages to a separate program.  That's the big change that should be done to improve the low-level part of Solaar, not turning magic command numbers into magic command names.

**Comment by MattHag** [comment]
Going that direction is even possible with the current interface. Nice, if there's a better interface.

> improve the low-level part of Solaar, not turning magic command numbers into magic command names.
No, not with magic command names, but with register_names for example. That makes code self-explanatory, and other uses of that register are much easier to find.

**Comment by MattHag** [comment]
If a call like `self._device.feature_request(self.feature, 0x70, 0x00)` is necessary multiple times, but naming registers doesn't make much sense to describe, then a partial can be helpful to accomplish.

This makes the function self documenting, it follows the DRY principle and makes the code more readable.
`signal_device_to_apply_changes = partial(self._device.feature_request, self.feature, 0x50, 0x00)

signal_device_to_apply_changes()
signal_device_to_apply_changes()`

**Pull request merged**